src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CurrentPeriodService.java

```diff
@@ -115,15 +117,15 @@ public String generateID(String value) {
     }
 
     public String createSelfLink(Transaction transaction, String companyAccountId) {
-        return transaction.getLinks().get(LinkType.SELF.getLink()) + "/"
-                + ResourceName.COMPANY_ACCOUNT.getName() + "/"
-                + companyAccountId + "/" + ResourceName.SMALL_FULL.getName() + "/"
-                + ResourceName.CURRENT_PERIOD.getName();
+        return transaction.getLinks().get(BasicLinkType.SELF.getLink()) + "/"
```

Comment:
again, when reading a link from a transaction we should technically use `TransactionLinkType`